profiles/ding2/modules/opensearch/opensearch.client.inc

```diff
@@ -516,6 +516,13 @@ function opensearch_execute(TingClientRequest $request) {
       if ($original_local_ids) {
         $request->setLocalIds($original_local_ids);
       }
+
+      // If response is empty array it means ther was no objects found for the
+      // passed ids. In this case we need to return NULL as expected by the
+      // caller.
+      if (empty($response)) {
+        $response = NULL;
```

Comment:
Should we return `NULL` (early) here rather than continue processing the empty response?

Reply:
It's from DDB Core and in the caching code, so i would like it to be 1:1 with core!